examples/piggy-bank/part1/src/lib.rs

```diff
@@ -74,12 +78,12 @@ fn piggy_smash<S: HasStateApi>(
 }
 
 /// View the state and balance of the piggy bank.
-#[receive(contract = "PiggyBank", name = "view")]
+#[receive(contract = "PiggyBank", name = "view", return_value = "ViewState")]
 fn piggy_view<S: HasStateApi>(
     _ctx: &impl HasReceiveContext,
     host: &impl HasHost<PiggyBankState, StateApiType = S>,
-) -> ReceiveResult<(PiggyBankState, Amount)> {
+) -> ReceiveResult<ViewState> {
     let current_state = *host.state();
     let current_balance = host.self_balance();
-    Ok((current_state, current_balance))
+    Ok(ViewState(current_state, current_balance))
```

Comment:
Not as elegant as the regular tuple, but I don't think the macro supports it

Reply:
Why do you say that?
```
#[receive(contract = "PiggyBank", name = "view", return_value = "(PiggyBankState, Amount)")]
```
works perfectly fine as I see it.

Reply:
I was mistaking, I tested it and most likely got another error and thought that the tuple was the culprit.